scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -763,5 +766,386 @@
+             (sched/filter-based-on-quota {test-user {:count 4, :cpus 20, :mem 6144}} user->usage queue))))))
+
+(deftest test-category->pending-jobs->category->considerable-jobs
+  (let [uri "datomic:mem://test-category-pending-jobs-category-considerable-jobs"
+        conn (restore-fresh-database! uri)
+        test-db (d/db conn)
+        test-user (System/getProperty "user.name")
+        group-ent-id (create-dummy-group conn)
+        job-1 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 3 :memory 2048))
+        job-2 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 13 :memory 1024))
+        job-3 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 7 :memory 4096))
+        job-4 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 11 :memory 1024))
+        job-5 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 5 :memory 2048 :gpus 2))
+        job-6 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 19 :memory 1024 :gpus 4))
+        category->pending-jobs {:normal [job-1 job-2 job-3 job-4], :gpu [job-5 job-6]}]
+
+    (testing "jobs inside usage quota"
+      (let [user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}}
+            user->quota {test-user {:count 10, :cpus 50, :mem 32768, :gpus 10}}
+            num-considerable 5]
+        (is (= {:normal [job-1 job-2 job-3 job-4], :gpu [job-5 job-6]}
+               (sched/category->pending-jobs->category->considerable-jobs
+                 (d/db conn) category->pending-jobs user->quota user->usage num-considerable)))))
+    (testing "some jobs inside usage quota"
+      (let [user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}}
+            user->quota {test-user {:count 5, :cpus 10, :mem 4096, :gpus 10}}
+            num-considerable 5]
+        (is (= {:normal [job-1], :gpu [job-5]}
+               (sched/category->pending-jobs->category->considerable-jobs
+                 (d/db conn) category->pending-jobs user->quota user->usage num-considerable)))))
+    (testing "some jobs inside usage quota - quota gpus ignored!"
+      (let [user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}}
+            user->quota {test-user {:count 5, :cpus 10, :mem 4096, :gpus 0}}
+            num-considerable 5]
+        (is (= {:normal [job-1], :gpu [job-5]}
+               (sched/category->pending-jobs->category->considerable-jobs
+                 (d/db conn) category->pending-jobs user->quota user->usage num-considerable)))))
+    (testing "all jobs exceed quota"
+      (let [user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}}
+            user->quota {test-user {:count 5, :cpus 3, :mem 4096, :gpus 10}}
+            num-considerable 5]
+        (is (= {:normal [], :gpu []}
+               (sched/category->pending-jobs->category->considerable-jobs
+                 (d/db conn) category->pending-jobs user->quota user->usage num-considerable)))))))
```

Comment:
Can we add a test where num-considerable actually truncates the amount of considerable jobs?